### PR TITLE
UPSTREAM: 31276: Update oc create success message when using --dry-run

### PR DESCRIPTION
**Pull request by juanvallejo**
UPSTREAM https://github.com/kubernetes/kubernetes/pull/31276

Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1369399

`oc create <resource> <name> --dry-run` provides a misleading success
message.

When commands such as `oc new-app node` are run with a
`--dry-run` flag, they make this clear by appending a `(dry run)`
string to the final output. `oc create <resource>  <name> --dry-run`
does not do this, providing a potentially misleading output.

This patch appends a "(DRY RUN)" string to the end of a successful
message of `oc create` subcommands that support the `--dry-run` flag.

`oc create quota quota --dry-run`
```
resourcequota "quota" created
```

`oc create quota quota --dry-run`
```
resourcequota "quota" created (dry run)
```

cc @fabianofranz

### CODE REVIEW

**Comment by juanvallejo** [comment]
[test]

**Comment by fabianofranz** [comment]
LGTM, this was much needed. @liggitt for approval.

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/expose.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 			return f.PrintObject(cmd, mapper, object, out)
 		}
 
-		cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, "exposed")
+		cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, cmdutil.GetDryRunFlag(cmd), "exposed")
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/expose.go]: we're already returning above:
```
if cmdutil.GetDryRunFlag(cmd) {
  return f.PrintObject(cmd, mapper, object, out)
}
```

should we be doing this above?
```
if cmdutil.GetDryRunFlag(cmd) {
  if len(cmdutil.GetFlagString(cmd, "output")) > 0 {
     return f.PrintObject(cmd, mapper, object, out)
  }
  cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, true, "exposed")
  return nil
}
```

and just passing `false` here?

Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/expose.go]: @liggitt 
> should we be doing this above?

That definitely works better. Thanks!

**Comment by liggitt** [comment]
quick glance LGTM, but not a 1.3 blocker

**Comment by juanvallejo** [comment]
re[test]

**Comment by juanvallejo** [comment]
[check](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/6176/) flaked on https://github.com/openshift/origin/issues/11057 re[test]

**Comment by juanvallejo** [comment]
[conformance](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6350/) test flaked on https://github.com/openshift/origin/issues/9548

[check](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/6229/) flaked  on https://github.com/openshift/origin/issues/11057
re[test]

**Comment by juanvallejo** [comment]
integration flaked on https://github.com/openshift/origin/issues/11240 re[test]

**Comment by juanvallejo** [comment]
test flaked on https://github.com/openshift/origin/issues/11442 re[test]

**Comment by juanvallejo** [comment]
test flaked on https://github.com/openshift/origin/issues/9203 re[test]

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**